mm2src/adex_cli/src/scenarios/init_mm2_cfg.rs

```diff
@@ -0,0 +1,276 @@
+use inquire::{validator::Validation, Confirm, CustomType, CustomUserError, Password, Text};
+use log::{error, info};
+use serde::Serialize;
+use std::net::Ipv4Addr;
+use std::ops::Not;
+use std::path::Path;
+
+use super::helpers;
+use super::inquire_extentions::{InquireOption, DEFAULT_DEFAULT_OPTION_BOOL_FORMATTER, DEFAULT_OPTION_BOOL_FORMATTER,
+                                OPTION_BOOL_PARSER};
+use common::password_policy;
+
+const DEFAULT_NET_ID: u16 = 7777;
+const DEFAULT_GID: &str = concat!("QA CLI ", env!("CARGO_PKG_VERSION"));
+const DEFAULT_OPTION_PLACEHOLDER: &str = "Tap enter to skip";
+
+pub fn init_mm2_cfg(cfg_file: &str) -> Result<(), ()> {
+    let mut mm2_cfg = Mm2Cfg::new();
+    info!("Start collecting mm2_cfg into: {cfg_file}");
+    mm2_cfg.inquire()?;
+    helpers::rewrite_json_file(&mm2_cfg, cfg_file)?;
+    info!("mm2_cfg has been writen into: {cfg_file}");
+
+    Ok(())
+}
+
+#[derive(Serialize)]
+pub struct Mm2Cfg {
+    pub gui: Option<String>,
+    pub net_id: Option<u16>,
+    pub rpc_password: Option<String>,
+    pub passphrase: Option<String>,
+    pub allow_weak_password: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub userhome: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub dbdir: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rpcip: Option<Ipv4Addr>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rpcport: Option<u16>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rpc_local_only: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub i_am_seed: Option<bool>,
+    #[serde(skip_serializing_if = "Vec::<Ipv4Addr>::is_empty")]
+    pub seednodes: Vec<Ipv4Addr>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub hd_account_id: Option<u64>,
+}
+
+impl Mm2Cfg {
+    pub fn new() -> Mm2Cfg {
+        Mm2Cfg {
+            gui: None,
+            net_id: None,
+            rpc_password: None,
+            passphrase: None,
+            allow_weak_password: None,
+            userhome: None,
+            dbdir: None,
+            rpcip: None,
+            rpcport: None,
+            rpc_local_only: None,
+            i_am_seed: None,
+            seednodes: Vec::<Ipv4Addr>::new(),
+            hd_account_id: None,
+        }
+    }
+
+    fn inquire(&mut self) -> Result<(), ()> {
+        self.inquire_gui()?;
+        self.inquire_net_id()?;
+        self.inquire_passphrase()?;
+        self.inquire_rpc_password()?;
+        self.inquire_allow_weak_password()?;
+        self.inquire_userhome()?;
+        self.inquire_dbdir()?;
+        self.inquire_rpcip()?;
+        self.inquire_rpcport()?;
+        self.inquire_rpc_local_only()?;
+        self.inquire_i_am_a_seed()?;
+        self.inquire_seednodes()?;
+        self.inquire_hd_account_id()?;
```

Comment:
If I had left all inquiries as a single method, it would have resulted in a huge, messy block of code. Each inquire_ method is quite complex and unique to isolate it in standalone method. Therefore, I don't see any conflict with the principles of clean code.

Is this remark is optional?

Reply:
Can't we create in-function generic function and use that? Defining function for each field doesn't seem too good and not an optimal way.

Reply:
These are already quire packed and all generic things are encapsulated inside the inquire crate. Each configuration part is inquired with its own help message, default, validator, placeholder and the like )))))

Reply:
Regardless adex cli and furthermore `adex-cli init` are not performace sensible - I've marked all the `inquire*` method as `inline`